common/match/matching.perf.ts

```diff
@@ -0,0 +1,339 @@
+    grade?: string;
+}
+
+const pupils = JSON.parse(readFileSync(__dirname + '/matching.perf.pupil.json', { encoding: 'utf-8' })) as HistoryEntry[];
+const students = JSON.parse(readFileSync(__dirname + '/matching.perf.student.json', { encoding: 'utf-8' })) as HistoryEntry[];
+
+async function computeOldMatchings(
+    requests: MatchRequest[],
+    offers: MatchOffer[],
+    matchesPerPupil: { [pupilId: string]: string[] },
+    matchesPerStudent: { [studentId: string]: string[] }
+) {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    const { match } = await import('corona-school-matching');
+
+    const helperByUUID = new Map();
+    const helpeeByUUID = new Map();
+    const requestByUUID = new Map();
+    const offerByUUID = new Map();
+
+    for (const request of requests) {
+        const uuid = `${request.pupilId}`;
+        if (helpeeByUUID.has(uuid)) {
+            helpeeByUUID.get(uuid).matchRequestCount += 1;
+        } else {
+            requestByUUID.set(uuid, request);
+            helpeeByUUID.set(uuid, {
+                id: request.pupilId,
+                uuid,
+                matchRequestCount: 1,
+                subjects: request.subjects.map(formattedSubjectToSubjectWithGradeRestriction),
+                createdAt: request.requestAt,
+                excludeMatchesWith: (matchesPerPupil[uuid] ?? []).map((uuid) => ({ uuid })),
+                state: request.state,
+                grade: request.grade,
+                matchingPriority: 1,
+                // firstMatchRequest: student.firstMatchRequest
+            });
+        }
+    }
+
+    for (const offer of offers) {
+        const uuid = `${offer.studentId}`;
+        if (helperByUUID.has(uuid)) {
+            helperByUUID.get(uuid).matchRequestCount += 1;
+        } else {
+            offerByUUID.set(uuid, offer);
+            helperByUUID.set(uuid, {
+                id: offer.studentId,
+                uuid,
+                matchRequestCount: 1,
+                subjects: offer.subjects,
+                createdAt: offer.requestAt,
+                excludeMatchesWith: (matchesPerStudent[uuid] ?? []).map((uuid) => ({ uuid })),
+                state: offer.state,
+                // firstMatchRequest: student.firstMatchRequest
+            });
+        }
+    }
+
+    const result = match([...helperByUUID.values()], [...helpeeByUUID.values()], {
+        balancingCoefficients: {
+            subjectMatching: 0.65,
+            state: 0.05,
+            waitingTime: 0.2,
+            matchingPriority: 0.1,
+        },
+    });
+
+    const matching: Matching = result.matches.map((it) => ({
+        request: requestByUUID.get(it.helpee.uuid),
+        offer: offerByUUID.get(it.helper.uuid),
+    }));
+
+    return matching;
+}
+
+describe('Real World Matching Performance', () => {
+    test.each([
+        // New Algorithm
+        [
+            'new',
+            1000,
+            {
```

Comment:
Using such a fixed time series as a benchmark is not so optimal - it can easily lead to overfitting onto this time series. Some sort of shuffling of the input would be good, but on the other hand that might loose some time effects that are actually there (e.g. pupils / students signing up in bulk after marketing campaigns)